pkg/config/store.go

```diff
@@ -78,6 +78,10 @@ type ChangeNotifier interface {
 	// The client should issue ReadChangeLog to see what has changed if the call is available.
 	// else it should re-read the store, perform diff and apply changes.
 	RegisterStoreChangeListener(s StoreListener)
+
+	// IsStoreChangeAvailable returns true only when the backend store is configured
+	// to emit change notifications.
+	IsStoreChangeAvailable() bool
```

Comment:
Ok, I thought that was your intended usecase (deciding programmatically whether to register a listener or do the polling yourself).

> so that manager can fall back to loop method when the change events are not enabled on the current storage.

That's exactly what I want us to avoid: it should not be the responsibility of the manager to choose how it interacts with the config store. The `store` package provides interfaces abstracting over being notified when changes occur and reading those changes when they do occur. It is the responsibility the thing implementing the `store` interfaces to work out the nitty gritty of _how_ it produces change notifications. If the manager has to decide then we haven't gotten the abstractions right. Instead, we should have an active polling implementation of `ChangeNotifier` as well as an implementation using notification mechanism native to the backend store. Then the operator selects the notification method when constructing a Mixer binary.